spec/std/enumerable_spec.cr

```diff
@@ -92,6 +92,94 @@ describe "Enumerable" do
     end
   end
 
+  describe "each_chunk" do
+    it "works" do
+      [1].each_chunk { true }.to_a.should eq [{true, [1]}]
+      [1, 2].each_chunk { false }.to_a.should eq [{false, [1, 2]}]
+      [1, 1, 2, 3, 3].each_chunk(&.itself).to_a.should eq [{1, [1, 1]}, {2, [2]}, {3, [3, 3]}]
+      [1, 1, 2, 3, 3].each_chunk(&.<=(2)).to_a.should eq [{true, [1, 1, 2]}, {false, [3, 3]}]
+      (0..10).each_chunk(&./(3)).to_a.should eq [{0, [0, 1, 2]}, {1, [3, 4, 5]}, {2, [6, 7, 8]}, {3, [9, 10]}]
+    end
+
+    pending "work with class" do
+      [1, 1, 2, 3, 3].each_chunk(&.class).to_a.should eq [{Int32, [1, 1, 2, 3, 3]}]
+    end
```

Comment:
nil is fixed, but now this not compiled because try to create `uninitialized Int32.class`